extensions/amp-form/0.1/amp-form.js

```diff
@@ -488,14 +505,51 @@ export class AmpForm {
       dev().error(TAG, 'Form submit service failed: %s', e);
     }
 
-    const varSubsFields = this.getVarSubsFields_();
-    if (this.xhrAction_) {
-      this.handleXhrSubmit_(varSubsFields, trust);
-    } else if (this.method_ == 'POST') {
-      this.handleNonXhrPost_();
-    } else if (this.method_ == 'GET') {
-      this.handleNonXhrGet_(varSubsFields);
+    // Get our special fields
+    const varSubsFields =
+      this.getVarSubsFields_();
+    const asyncInputs =
+      this.form_.getElementsByClassName(AsyncInputClasses.ASYNC_INPUT);
+
+    // Do any assertions we may need to do
+    // For NonXhrGET
+    // That way we can determine if
+    // we can submit synchronously
+    if (!this.xhrAction_ && this.method_ == 'GET') {
+      this.assertSsrTemplate_(false, 'Non-XHR GETs not supported.');
+      this.assertNoSensitiveFields_();
+
+      // If we have no async inputs, we can just submit synchronously
+      if (asyncInputs.length === 0) {
+
+        for (let i = 0; i < varSubsFields.length; i++) {
+          this.urlReplacement_.expandInputValueSync(varSubsFields[i]);
+        }
+
+        this.handleNonXhrGet_(event);
+        return Promise.resolve();
+      }
+
+      event.preventDefault();
     }
+
+    // Set ourselves to the SUBMITTING State
+    this.setState_(FormState.SUBMITTING);
+
+    // Promises to run before submitting the form
+    const presubmitPromises = [];
+    presubmitPromises.push(this.doVarSubs_(varSubsFields));
+    iterateCursor(asyncInputs, asyncInput => {
+      presubmitPromises.push(this.getValueForAsyncInput_(asyncInput));
+    });
+
+    return this.waitOnPromisesOrTimeout_(
+        presubmitPromises,
+        SUBMIT_TIMEOUT
```

Comment:
I noticed one thing here. The wait for var substitution is 100ms, while the wait for async input is 10000ms. However we are combining the wait for varSubsFields and asyncInput here, and the timeout is 10000ms for all. Which could lead to a 10s wait for var substitution. 

 to @cvializ to decide whether this is ok or not.

Reply:
The variable substitution will still timeout at 100ms since it also calls `waitOnPromisesOrTimeout`. 10000ms should be fine to begin with, and we can shorten it later if it ends up being too long.